### DISCUSSION TITLE
Storybook

### DISCUSSION THREAD

**Discussion by nus3** [freshframework/fresh#542] [Q&A]
Does Fresh plan to offer component-driven development tools like Storybook?

**Comment by excessivecoding** [comment]
It's not included but as Fresh is only using Preact under the hood. so you could always make a custom setup with Storybook and Preact

**Reply by nus3** [reply]
Thanks for the answer!

> so you could always make a custom setup with Storybook and Preact

If I want to use Storybook with Fresh, can I use Deno's Node compatibility mode?
Or should I use npm for Storybook?

**Reply by excessivecoding** [reply]
I'm not experienced enough with Storybook nor with Deno's Node compat lib, but I'm guessing it could not be possible due to Storybook dependencies to webpack, babel etc. which are not compatible afaik.

I would personally go with the NPM / Node approach to limit friction.

You might want to make some research on the subject, though, as I'm really not a reliable source of information on those matters.

**Reply by nus3** [reply]
Thanks!
I will try to use npm for Storybook.

**Comment by CarcajadaArtificial** [comment]
Hey OP, I'm going for something similar, an integration for my Fresh components (and maybe islands) to Storybook. I'm having trouble including the dependency to even start using it. Were you successful in your integration attempt?
Thanks

**Comment by CAYdenberg** [comment]
Hi everyone! I've written a simple storybook implementation for Fresh: https://deno.land/x/denostories
